src/main/java/pl/put/poznan/sqc/logic/Interaction.java

```diff
@@ -3,27 +3,125 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import pl.put.poznan.sqc.rest.ScenarioQualityCheckerController;
+import pl.put.poznan.sqc.scenario.Actor;
 import pl.put.poznan.sqc.scenario.Scenario;
+import pl.put.poznan.sqc.scenario.Step;
 
+import java.util.ArrayList;
+
+/**
+ * Interaction counting class implementation
+ */
 public class Interaction implements ScenarioInterface {
 
+    /**
+     * logging support param in LoggerFactory type
+     */
     private static final Logger logger = LoggerFactory.getLogger(ScenarioQualityCheckerController.class);
 
-    public String getTitle() {
-        return title;
-    }
-
+    /**
+     * Title of return scenario
+     */
     private final String title;
 
+    /**
+     * List of steps with interactions between actors
+     */
+    private ArrayList<String> StepsWithInteractions;
+
+    /**
+     * Getter for StepsWithInteractions private variable
+     *
+     * @return variable StepsWithInteractions List of Strings value
+     */
+    public ArrayList<String> getStepsWithInteractions(){return StepsWithInteractions;}
+
+    /**
+     * Getter for title private variable
+     *
+     * @return variable title String value
+     */
+    public String getTitle() { return title;}
+
+    /**
+     * Interaction class constructor
+     *
+     * @param scenario in Scenario type
+     */
     public Interaction(Scenario scenario) {
+
         this.title = scenario.getTitle();
+        this.StepsWithInteractions = new ArrayList<>();
+
+    }
+
+    /**
+     *  Interactions checking method fot substeps
```

Comment:
fot -> for
Dodałbym coś o rekurencyjności funkcji np.  'Interactions checking method for sub-steps and its sub-steps recursively'

Reply:
Wszystko poprawione według zaleceń.